### ISSUE TITLE
BeforeInsert not triggered on bulk save, despite all items being instances of entity

### ISSUE THREAD

**Issue by cooperfrench95** [typeorm/typeorm#5785]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[x] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

```javascript
const processed = await Promise.all(assigns.map(async a => {
    const assign = new Assignment()
    const toReturn = Object.assign(assign, a)
    return toReturn
  }))
  console.log(processed) // all instanceOf assignment
  await getConnection()
    .getRepository('Assignment')
    .save(processed, { chunk: 999, transaction: true, listeners: true })
```

entity:

```javascript
@Entity()
class Assignment {
   ....
  @Column({ type: 'integer' }, { nullable: true })
  start = undefined;

  @Column({ type: 'integer' }, { nullable: true })
  end = undefined;
   ....
  @BeforeInsert()
  parseJsonFields() {
    Object.keys(this).forEach((relAttr) => {
      console.log(relAttr) // does not log
      if ([ 'start', 'end'].includes(relAttr)) {
        const attr = this[relAttr]
        switch(relAttr) {
          ...
          case 'start':
          case 'end':
            this[relAttr] = moment(attr).unix() // this doesn't get run
            break
          default:
            break
        }
      }
    })
  }
```

Result in DB is that start and end are time strings instead of ints.

**Comment by cooperfrench95** [comment]
Solved, thanks to #5493

The issue was that

```javascript
const assign = Object.assign(new Assignment(), data)
```

does **not** create the instance of the entity properly, despite `assign` being `instanceof Assignment`.
The following works:

```javascript
const assign = await getManager().create('Assignment', data)
```

**Comment by ohohua** [comment]
good!
